src/main/java/mate/academy/spring/service/impl/RoleServiceImpl.java

```diff
@@ -0,0 +1,27 @@
+package mate.academy.spring.service.impl;
+
+import java.util.NoSuchElementException;
+import mate.academy.spring.dao.RoleDao;
+import mate.academy.spring.model.Role;
+import mate.academy.spring.service.RoleService;
+import org.springframework.stereotype.Service;
+
+@Service
+public class RoleServiceImpl implements RoleService {
+    private final RoleDao roleDao;
+
+    public RoleServiceImpl(RoleDao roleDao) {
+        this.roleDao = roleDao;
+    }
+
+    @Override
+    public Role add(Role role) {
+        return roleDao.add(role);
+    }
+
+    @Override
+    public Role getByName(String roleName) {
+        return roleDao.getByName(roleName).orElseThrow(() ->
```

Comment:
From common mistakes:
Be careful in getByName() method on dao layer! You should pass enum to query (not String), otherwise you won’t be able to get Role by name.